File: backend/src/main/java/com/socialmeet/backend/dto/WalletDTO.java
package com.socialmeet.backend.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 钱包信息DTO
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class WalletDTO {

    private Long userId;
    private BigDecimal balance;
    private BigDecimal totalRecharge;
    private BigDecimal totalConsume;
    private Integer transactionCount;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime lastTransactionAt;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;
}

File: app/src/main/java/com/example/myapplication/dto/WalletDTO.kt
package com.example.myapplication.dto

import java.math.BigDecimal

data class WalletDTO(
    val id: Long? = null,
    val userId: Long? = null,
    val balance: BigDecimal? = null,
    val frozenAmount: BigDecimal? = null,
    val currency: String? = null,
    val availableBalance: BigDecimal? = null,
    val createdAt: String? = null,
    val updatedAt: String? = null
)
